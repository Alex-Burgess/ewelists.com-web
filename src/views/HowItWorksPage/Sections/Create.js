import React from 'react';
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// core components
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Button from "components/Buttons/Button.js";

import styles from "assets/jss/material-kit-pro-react/views/howItWorksPage/createStyle.js";
const useStyles = makeStyles(styles);

export default function LandingCreate(props) {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <GridContainer justify="center">
        <GridItem xs={12} sm={8} md={8} className={classes.textCenter}>
          <h3 className={classes.subTitle}>
            Ready to try Ewelists?
          </h3>
          <Button size="lg" color="danger" href="/signup" data-cy="button-get-started">
            Get Started
          </Button>
        </GridItem>
      </GridContainer>
    </div>
  );
}
