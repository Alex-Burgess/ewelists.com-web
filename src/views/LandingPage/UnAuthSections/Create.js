import React, { useState, useEffect } from 'react';
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// @material-ui/icons
// core components
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Button from "components/CustomButtons/Button.js";

import laptop from "assets/img/laptop-mockup.png";
// import sheeptoy from "assets/img/sheep-toy1.jpg";

import styles from "assets/jss/custom/views/landingPage/createStyle.js";
const useStyles = makeStyles(styles);

export default function LandingCreate(props) {
  const classes = useStyles();

  const [desktop, setDesktop] = useState(true);

  useEffect( () => {
    function updateDimensions() {
      if (window.innerWidth < 600){
        setDesktop(false);
      } else {
        setDesktop(true);
      }
    };

    window.addEventListener('resize', updateDimensions);
    updateDimensions();
  }, []);


  return (
    <div className={classes.section}>
      <div className={classes.container}>
        <GridContainer>
          <GridItem xs={12} sm={5} md={5}>
            <h1 className={classes.title}>Create your Baby Gift List</h1>
            <h4 className={classes.description}>
              Ewelists helps new and expecting parents get the gifts they
              need for their new arrival. Create a gift list, which you can
              share with friends and family, for free.
            </h4>
            <br />
            { desktop
              ? null
              : <div className={classes.imgContainer}>
                  <img src={laptop} alt="..." />
                </div>
            }
            <div className={classes.buttonContainer}>
              <Button
                color="danger"
                size="lg"
                href="/signup"
                target="_blank"
              >
                Create Your List - It's Free!
              </Button>
            </div>
          </GridItem>
          <GridItem xs={12} sm={7} md={7}>
            { desktop
              ? <div className={classes.imgContainer}>
                  <img src={laptop} alt="..." />
                </div>
              : null
            }
          </GridItem>
        </GridContainer>
      </div>
    </div>
  );
}
