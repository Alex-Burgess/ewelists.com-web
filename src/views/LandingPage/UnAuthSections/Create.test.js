import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Create from './Create';

describe('Tests for Create section.', () => {
  test('Renders create section', () => {
    const tree = TestRenderer
      .create(<Create />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Renders create section on mobile device width', () => {
    global.innerWidth = 375;

    const tree = TestRenderer
      .create(<Create />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
