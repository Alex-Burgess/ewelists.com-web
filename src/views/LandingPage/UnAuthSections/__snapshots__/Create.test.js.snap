// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Create section. Renders create section 1`] = `
<div
  className="makeStyles-section-3"
>
  <div
    className="makeStyles-container-1"
  >
    <div
      className="MuiGrid-root makeStyles-grid-8  MuiGrid-container"
    >
      <div
        className="MuiGrid-root makeStyles-grid-112  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
      >
        <h1
          className="makeStyles-title-4"
        >
          Create your Baby Gift List
        </h1>
        <h4
          className="makeStyles-description-2"
        >
          Ewelists helps new and expecting parents get the gifts they need for their new arrival. Create a gift list, which you can share with friends and family, for free.
        </h4>
        <br />
        <div
          className="makeStyles-buttonContainer-7"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-113 makeStyles-lg-138 makeStyles-danger-120"
            href="/signup"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            target="_blank"
          >
            <span
              className="MuiButton-label"
            >
              Create Your List - It's Free!
            </span>
          </a>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-grid-112  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7 MuiGrid-grid-md-7"
      >
        <div
          className="makeStyles-imgContainer-5"
        >
          <img
            alt="..."
            src="laptop-mockup.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests for Create section. Renders create section on mobile device width 1`] = `
<div
  className="makeStyles-section-3"
>
  <div
    className="makeStyles-container-1"
  >
    <div
      className="MuiGrid-root makeStyles-grid-8  MuiGrid-container"
    >
      <div
        className="MuiGrid-root makeStyles-grid-112  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
      >
        <h1
          className="makeStyles-title-4"
        >
          Create your Baby Gift List
        </h1>
        <h4
          className="makeStyles-description-2"
        >
          Ewelists helps new and expecting parents get the gifts they need for their new arrival. Create a gift list, which you can share with friends and family, for free.
        </h4>
        <br />
        <div
          className="makeStyles-imgContainer-5"
        >
          <img
            alt="..."
            src="laptop-mockup.png"
          />
        </div>
        <div
          className="makeStyles-buttonContainer-7"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-113 makeStyles-lg-138 makeStyles-danger-120"
            href="/signup"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            target="_blank"
          >
            <span
              className="MuiButton-label"
            >
              Create Your List - It's Free!
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-grid-112  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7 MuiGrid-grid-md-7"
      />
    </div>
  </div>
</div>
`;
